### PR TITLE
#4313 [Manager] Unavailable nodes should NOT be selectable

### PR DESCRIPTION
**Pull request by konekoya**
### Checklist
- [x] [Review Contribution Guidelines](https://ohara.readthedocs.io/en/latest/contributing.html)
- [x] Add Reviewers (see [Authors](https://github.com/oharastream/ohara#ohara-team))
- [x] Add Assignees (of course, you are the assignee by default!)
- [x] DON'T set a Milestone
- [x] DON'T add any labels (PR labels will be added automatically by Jenkins)
- [x] Link to an issue(You can do so by using the required by keyword: required by #4313)
- [ ] Add unit tests (or please explain why this PR is not worth having new tests)
- [x] Pass QA (Sometimes you may encounter an existing flaky test which would obstruct you from getting QA passed. If the failed tests are unrelated, you can add a comment in the PR thread)

### CODE REVIEW

File: ohara-manager/client/src/components/common/Progress/Percentage.js [hunk 1]
```diff
@@ -27,7 +27,7 @@ const Percentage = (values, key) => {
   }
   if (values.indexOf(key) < 0) {
     if (values.indexOf('null') > -1) {
-      return 'NaN';
+      return 'None';
```
Comment (konekoya) [inline ohara-manager/client/src/components/common/Progress/Percentage.js]: Correct wording

cc @vitojeng

File: ohara-manager/client/src/components/Node/NodeDialog.js [hunk 2]
```diff
@@ -172,9 +173,13 @@ const NodeDialog = () => {
...
 
-  const blockedRows = rows.filter(row => {
-    return blockedNodes.map(node => node.hostname).includes(row.name);
-  });
+  const blockedRows = rows.filter(row =>
+    blockedNodes.map(node => node.hostname).includes(row.name),
+  );
+
+  const unavailableRows = rows.filter(
+    row => row.state.toUpperCase() === state.unavailable,
```
Comment (konekoya) [inline ohara-manager/client/src/components/Node/NodeDialog.js]: We capitalize word in the prior step, and so we need to do a safer check here with another string transformation

**Comment by konekoya** [comment]
retry manager-e2e

**Comment by konekoya** [comment]
retry manager-e2e

**Comment by konekoya** [comment]
retry manager-e2e

File: ohara-manager/client/src/components/common/Progress/Percentage.js (hunk 1 above)
Comment (vitojeng) [reply ohara-manager/client/src/components/common/Progress/Percentage.js]: @konekoya Thanks!

Comment (jackyoh) [reply ohara-manager/client/src/components/common/Progress/Percentage.js]: 上面的 if 應該可以寫成 if (values.indexOf(key) < 0 && values.indexOf('null') > -1) 避免太多巢狀

Comment (konekoya) [reply ohara-manager/client/src/components/common/Progress/Percentage.js]: Good point @jackyoh :)

Comment (konekoya) [reply ohara-manager/client/src/components/common/Progress/Percentage.js]: @jackyoh Sorry, it's in fact must be nesting the if statements like this as there are two different branches here.

First, check if the key is matching: `values.indexOf(key) < 0`
Second, check if the values contain any `null`s: `values.indexOf('null') > -1` and return different values

It's kind of messy, but helpfully you understand what I'm talking about... 😅

Comment (jackyoh) [reply ohara-manager/client/src/components/common/Progress/Percentage.js]: 了解了，Thank you.

**Comment by chia7712** [comment]
retry manager-e2e

**Comment by konekoya** [comment]
retry manager-e2e

**Comment by saivirtue** [comment]
I've traced the e2e tests in #4049, It's OK to merge this PR after the build error fixed :)

**Comment by konekoya** [comment]
retry manager-e2e

**Comment by konekoya** [comment]
retry manager-e2e

**Comment by konekoya** [comment]
> I've traced the e2e tests in #4049, It's OK to merge this PR after the build error fixed :)

Just about the right time 🙏 

@chia7712 All checks passed :)

**Pull request merged**